modules/packages/LinearAlgebra.chpl

```diff
@@ -927,6 +927,28 @@ proc trace(A: [?D] ?eltType) {
   return trace;
 }
 
+/* Return the Kronecker Product of matrix ``A`` and matrix ``B``.
+   If the size of A is ``x * y`` and of B is ``a * b`` then size of the resulting
+   matrix will be ``(x * a) * (y * b)`` */
+proc kron(A: [?ADom] ?eltType, B: [?BDom] eltType) {
+  if ADom.rank != 2 || BDom.rank != 2 then compilerError("Rank size not 2");
+  const (rowA, colA) = A.shape;
```

Comment:
Thinking on this more, this implementation makes the assumption that `A` and `B` have 0-based domains, i.e. `{0..#rowA, 0..colA}`. The `LinearAlgebra` library is intended to be generic w.r.t. domain indices. For this reason I suggest we do a reindexing for `A` and `B` so that we can access them as if they were 0-based within the function. Also, I now notice that we can combine the nested for loops into 2 loops over the 2-dimensional domains, using the `forall` in the outer loop. Here's what I suggest for these changes:

```chpl

  const AzDom = {0..#rowA, 0..#colA},
        BzDom = {0..#rowB, 0..#colB};

  ref Az = A.reindex(AzDom),
      Bz = B.reindex(BzDom);

  var C = Matrix(rowA*rowB, colA*colB, eltType=eltType);

  forall (i,j) in AzDom {
    const stR = i*rowB,
          stC = j*colB;
    for (k,l) in BzDom {
      C[stR+k, stC+l] = Az[i, j]*Bz[k, l];
    }
  }
```

Reply:
ok, this sounds interesting, I change it, thanks